src/anndata/compat/__init__.py

```diff
@@ -100,6 +100,32 @@ def __repr__():
             return "mock dask.array.core.Array"
 
 
+if find_spec("xarray") or TYPE_CHECKING:
+    import xarray
+    from xarray import DataArray as XArray
```

Comment:
```suggestion
    from xarray import DataArray as XDataArray
```
for consistency (and elsewhere) - i like the prefix a lot, great call